### PR TITLE
[Top Bar Feature] Remove all references to `dynamicTopBarAndReframe` feature

### PR DESCRIPTION
**Pull request by mrcthms**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-internal/issues/481

As part of the clean up around the dynamicTopBarAndReframe cleanup, we want to revert all updates for this feature within `@shopify/polaris` back to its initial state.

### WHAT is this pull request doing?

Remove all references to code changes around the `dynamicTopBarAndReframe` and ensure behaviour is reverted back to how it was before the feature was introduced.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

Flow sandbox with snapshot: https://flow.polaris-top-bar-test.marc-thomas.eu.spin.dev/dev-sandbox/overview/0190a740-3aad-7a63-a48b-30c3578f9a83

### 🎩 checklist

- [x] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Frame/Frame.tsx [hunk 1]
```diff
@@ -404,17 +348,6 @@ class FrameInner extends PureComponent<CombinedProps, State> {
     );
   };
 
-  private setBodyStyles = () => {
```
Comment (laurkim) [inline polaris-react/src/components/Frame/Frame.tsx:407]: I think this function is still needed for the conditionals that aren't checking for `dynamicTopBarAndReframe`, i.e., lines 408-410 and 414.

Comment (mrcthms) [reply polaris-react/src/components/Frame/Frame.tsx:407]: @laurkim I can't find reference to `setBodyStyles` in the [Frame](https://github.com/Shopify/polaris/blob/955bc3843c83d8c94a88ab131a2e6759ea9981c6/polaris-react/src/components/Frame/Frame.tsx) file, was it there or somewhere elsewhere you were seeing it?

Comment (laurkim) [reply polaris-react/src/components/Frame/Frame.tsx:407]: Oh no you're right, I just looked at the PR where the function was being added and looks like it's safe to remove 👍

**Review by laurkim — APPROVED** [review]

**Pull request merged**